codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -131,18 +131,24 @@ final boolean isClosed() {
             promise.trySuccess(null);
             return promise;
         }
+
         Iterator<PrimitiveEntry<Http2Stream>> itr = streamMap.entries().iterator();
         // We must take care while iterating the streamMap as to not modify while iterating in case there are other code
         // paths iterating over the active streams.
         if (activeStreams.allowModifications()) {
-            while (itr.hasNext()) {
-                DefaultStream stream = (DefaultStream) itr.next().value();
-                if (stream.id() != CONNECTION_STREAM_ID) {
-                    // If modifications of the activeStream map is allowed, then a stream close operation will also
-                    // modify the streamMap. Pass the iterator in so that remove will be called to prevent concurrent
-                    // modification exceptions.
-                    stream.close(itr);
+            activeStreams.incrementPendingIterations();
```

Comment:
> can we just call forEachActiveStream instead

Unfortunately not. The original motivation for this method was because we didn't close *all* the streams. There can be streams that are not active and thus not in the `activeStreams` collection that still must be closed.